### ISSUE TITLE
Implementation of EvoNorm gives NaN loss

### ISSUE THREAD

**Issue by digantamisra98** [knighton/evo-norm#1]
Hi. 
I tried implementing EvoNorm as per the paper. Firstly, the pseudo code presented in the paper has an error in the group_std function as after getting moments across dimension [2,3,4], it is impossible to reshape it back to the original shape of the input tensor. However, I did solve it using the expand function in PyTorch but now I'm getting NaN loss in training. Could you maybe point out what's the issue with my implementation. Link - https://github.com/digantamisra98/EvoNorm/blob/master/evonorm2d.py
Thanks!

**Comment by digantamisra98** [comment]
Update: I did fix the NaN loss error. S0 is working as intended, however I am facing shape mismatch error in the running variance calculation in B0 training mode version, probably due to overwriting. You can take a look at the updated code here:https://github.com/digantamisra98/EvoNorm/blob/master/evonorm2d.py

**Comment by knighton** [comment]
Hi, sorry I've been pretty busy, just seeing this now.  How goes?

**Comment by digantamisra98** [comment]
Fixed
